test/test_parse_osmchange_input.cpp

```diff
@@ -481,401 +255,221 @@ void test_node() {
-  }
+TEST_CASE("Create relation, member type missing", "[osmchange][relation][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><create><relation changeset="972" id="-1"><member role="stop" ref="-1"/></relation></create></osmChange>)"), http::bad_request);
+}
 
-  // Invalid ref
-  try {
-    process_testmsg(
-        R"(<osmChange><create><relation changeset="972" id="-1"><member type="node" ref="a" role="stop"/></relation></create></osmChange>)");
-    throw std::runtime_error("test_relation::005: Expected exception");
-  } catch (http::exception &e) {
-    if (e.code() != 400)
-      throw std::runtime_error("test_relation::005: Expected HTTP 400");
-  }
+TEST_CASE("Create relation, invalid member type", "[osmchange][relation][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><create><relation changeset="972" id="-1"><member type="bla" role="stop" ref="-1"/></relation></create></osmChange>)"), http::bad_request);
+}
 
-  // Invalid zero ref
-  try {
-    process_testmsg(
-        R"(<osmChange><create><relation changeset="972" id="-1"><member type="way" ref="0" role="stop"/></relation></create></osmChange>)");
-    throw std::runtime_error("test_relation::006: Expected exception");
-  } catch (http::exception &e) {
-    if (e.code() != 400)
-      throw std::runtime_error("test_relation::006: Expected HTTP 400");
-  }
+TEST_CASE("Create relation, invalid member ref", "[osmchange][relation][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><create><relation changeset="972" id="-1"><member type="node" ref="a" role="stop"/></relation></create></osmChange>)"), http::bad_request);
+}
 
-  // ref too large
-  try {
-    process_testmsg(
-        R"(<osmChange><create><relation changeset="972" id="-1">
-           <member type="relation" ref="99999999999999999999999999999999" role="stop"/>
-           </relation></create></osmChange>)");
-    throw std::runtime_error("test_relation::007: Expected exception");
-  } catch (http::exception &e) {
-    if (e.code() != 400)
-      throw std::runtime_error("test_relation::007: Expected HTTP 400");
-  }
+TEST_CASE("Create relation, invalid member ref zero", "[osmchange][relation][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><create><relation changeset="972" id="-1"><member type="way" ref="0" role="stop"/></relation></create></osmChange>)"), http::bad_request);
+}
 
+TEST_CASE("Create relation, member ref too large", "[osmchange][relation][xml]") {
+  REQUIRE_THROWS_AS(process_testmsg(
+    R"(<osmChange><create><relation changeset="972" id="-1">
+           <member type="relation" ref="99999999999999999999999999999999" role="stop"/>
+           </relation></create></osmChange>)"), http::bad_request);
+}
 
-  // Member: role may have up to 255 unicode characters
-  for (int i = 1; i <= 256; i++) {
+TEST_CASE("Create relation, role with <= 255 unicode characters", "[osmchange][node][xml]") {
```

Comment:
There might be a typo/copy&paste issue here. I would expect `[osmchange][relation][xml]`